core/js/src/main/scala/zio/ZIOPlatformSpecific.scala

```diff
@@ -16,6 +16,40 @@
 
 package zio
 
-private[zio] trait ZIOPlatformSpecific[-R, +E, +A]
+import scala.scalajs.js
+import scala.scalajs.js.{ Promise => JSPromise }
 
-private[zio] trait ZIOCompanionPlatformSpecific
+private[zio] trait ZIOPlatformSpecific[-R, +E, +A] { self: ZIO[R, E, A] =>
+
+  /**
+   * Converts the current `ZIO` to a Scala.js promise.
+   */
+  def toPromiseJS(implicit ev: E <:< Throwable): URIO[R, JSPromise[A]] =
+    toPromiseJSWith(ev)
+
+  /**
+   * Converts the current `ZIO` to a Scala.js promise and maps the
+   * error type with `f`.
+   */
+  def toPromiseJSWith(f: E => Throwable): URIO[R, JSPromise[A]] =
+    self.mapError(f).fold(JSPromise.reject, JSPromise.resolve[A](_))
```

Comment:
I suggest to take possible defects in as well:
```suggestion
    self.foldCause(c => JSPromise.reject(c.squashWith(f)), JSPromise.resolve[A](_))
```